spec/unit/network/http/api/v3/indirected_routes_spec.rb

```diff
@@ -198,7 +207,12 @@ def a_request_that_searches(key, request = {})
     end
 
     it "should include the environment in the query string of the URI" do
-      handler.class.request_to_uri(request).should == "#{url_prefix}/foo/with%20spaces?environment=myenv&foo=bar"
+      handler.class.request_to_uri(request).should == "#{master_url_prefix}/foo/with%20spaces?environment=myenv&foo=bar"
+    end
+
+    it "should include the correct url prefix if it is a ca request" do
+      request.stubs(:indirection_name).returns("certificate")
+      handler.class.request_to_uri(request).should == "#{ca_url_prefix}/certificate/with%20spaces?environment=myenv&foo=bar"
```

Comment:
No, environment is required for all indirector requests.  Even the ones that don't use it for anything.  (Spoiler alert: we should not carry that over to v2 of the CA API :) )